infrastructure/lib/constructs/authentication.ts

```diff
@@ -215,22 +220,17 @@ export class CognitoAuthentication extends cdk.Construct {
         requireDigits: true,
         requireLowercase: true,
         requireSymbols: true,
-        requireUppercase: true,
+      },
+      customAttributes: {
+        groups: new cognito.StringAttribute({ mutable: true, maxLen: 2000 }),
+        dod_id: new cognito.StringAttribute({ mutable: true }),
```

Comment:
```suggestion
        dod_id: new cognito.StringAttribute({ minLen: 10, maxLen: 10, mutable: false }),
```

Feels like it'd enforce constraints a little better. But since we can't modify the field later, maybe we don't want to do that.